zeppelin-zengine/src/main/java/org/apache/zeppelin/search/LuceneSearch.java

```diff
@@ -51,43 +56,48 @@
 import org.apache.lucene.search.highlight.TextFragment;
 import org.apache.lucene.search.highlight.TokenSources;
 import org.apache.lucene.store.Directory;
-import org.apache.lucene.store.RAMDirectory;
+import org.apache.lucene.store.MMapDirectory;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
 import org.apache.zeppelin.notebook.Note;
 import org.apache.zeppelin.notebook.Paragraph;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.base.Joiner;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Lists;
-
 /**
  * Search (both, indexing and query) the notebooks using Lucene.
  *
- * Query is thread-safe, as creates new IndexReader every time.
- * Index is thread-safe, as re-uses single IndexWriter, which is thread-safe.
+ * <p>Query is thread-safe, as creates new IndexReader every time. Index is thread-safe, as re-uses
```

Comment:
hmm, why `<p>`?

Reply:
Formatter add <p> to affect <br /> in my guess. I'll remove it.